lib/model/content.dart

```diff
@@ -1107,7 +1107,7 @@ class _ZulipContentParser {
     for (final node in nodes) {
       // We get a bunch of newline Text nodes between paragraphs.
       // A browser seems to ignore these; let's do the same.
-      if (node is dom.Text && (node.text == '\n')) continue;
+      if (node is dom.Text && RegExp(r'^\n+$').hasMatch(node.text)) continue;
```

Comment:
This parsing code is a hot path, so I try to stay fairly paranoid about performance here.

(A bit more discussion here — I should probably add this point to the README or a style guide:
https://github.com/zulip/zulip-flutter/pull/587#discussion_r1539898705
https://github.com/zulip/zulip-flutter/pull/587#discussion_r1546882850
)

As part of that, instead of making a RegExp object inside a parsing function, when we need one it's a `static final` member on the class. For examples, search for "RegExp" in this file.